agent/include/content_analysis/sdk/analysis_agent.h

```diff
@@ -96,6 +96,17 @@ class ContentAnalysisEvent {
   // for debugging.
   virtual std::string DebugString() const = 0;
 
+  // Helper class to handle the lifetime and access of print data.
+  class ScopedPrintHandle {
+   public:
+    static std::unique_ptr<ScopedPrintHandle> Create(
+        const ContentAnalysisRequest& request);
+
+    //virtual ~ScopedPrintHandle();
```

Comment:
It's important to have a virtual dtor since there will be subclasses that override dtor behaviour and callers only deal with the interface class.  Can likely do a `ScopedPrintHandle::~ScopedPrintHandle() = default;` in cc file.